CMakeLists.txt

```diff
@@ -67,84 +74,93 @@ if(ASIO_CPM_FETCHCONTENT_COMPAT OR NOT ASIO_USE_CPM)
   FetchContent_GetProperties(asio)
 endif()
 
-if(ASIO_ENABLE_INSTALL)
-  include(GNUInstallDirs) # for CMAKE_INSTALL_INCLUDEDIR
-endif()
-
+#
+# Download and add library
+#
 if(NOT asio_POPULATED AND NOT TARGET asio)
-
-  # Download ##
-
   if(ASIO_USE_CPM)
-    include(cmake/get_cpm.cmake)
+    # Include CPM only if not already initialized
+    if(NOT CPM_INITIALIZED)
+      include(cmake/get_cpm.cmake)
+    endif()
+
     CPMAddPackage(
       NAME asio
       GIT_REPOSITORY ${ASIO_REPOSITORY}
       GIT_TAG ${ASIO_TAG}
       DOWNLOAD_ONLY TRUE
     )
   else()
-    message(STATUS "Download asio from ${ASIO_REPOSITORY}:${ASIO_TAG}")
+    message(STATUS "Download Asio from ${ASIO_REPOSITORY}:${ASIO_TAG}")
     FetchContent_Populate(asio)
   endif()
 
-  # Find Version ##
+  # Find version
   file(STRINGS "${asio_SOURCE_DIR}/asio/include/asio/version.hpp" ASIO_VERSION
        REGEX "^#define ASIO_VERSION [0-9]+"
   )
   string(REGEX REPLACE "^#define ASIO_VERSION [0-9]+ // ([0-9.]+)$" "\\1" ASIO_VERSION
                        "${ASIO_VERSION}"
   )
-
   set(ASIO_VERSION
       ${ASIO_VERSION}
-      CACHE STRING "asio library version" FORCE
+      CACHE STRING "Asio library version" FORCE
   )
 
-  # Configure ##
-
-  add_library(asio STATIC src/asio.cpp)
+  # Library
+  add_library(asio INTERFACE)
   add_library(asio::asio ALIAS asio)
   target_include_directories(
-    asio PUBLIC $<BUILD_INTERFACE:${asio_SOURCE_DIR}/asio/include>
-                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+    asio SYSTEM INTERFACE $<BUILD_INTERFACE:${asio_SOURCE_DIR}/asio/include>
+                          $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   )
 
-  find_package(Threads)
-  target_link_libraries(asio PRIVATE Threads::Threads)
+  find_package(Threads REQUIRED)
+  target_link_libraries(asio INTERFACE Threads::Threads)
 
-  target_compile_features(asio PUBLIC cxx_std_11)
+  set_target_properties(
+    asio
+    PROPERTIES CXX_STANDARD 11
+               CXX_STANDARD_REQUIRED ON
+               CXX_EXTENSIONS OFF
+  )
   target_compile_definitions(
-    asio PUBLIC -DASIO_STANDALONE -DASIO_SEPARATE_COMPILATION -DASIO_NO_DEPRECATED
+    asio INTERFACE ASIO_STANDALONE $<$<BOOL:${ASIO_NO_DEPRECATED}>:ASIO_NO_DEPRECATED>
```

Comment:
if you want to install it as header only library, you have to install the ipp files too!

Reply:
Please see PR https://github.com/OlivierLDff/asio.cmake/pull/35
This has been already fixed in that PR